engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -556,24 +556,13 @@ private void setupStringCodecs()
 
   private void setupWsClient()
   {
-    String gatewayAddress = plan.getLogicalPlan().getValue(LogicalPlan.GATEWAY_CONNECT_ADDRESS);
-    boolean gatewayUseSsl = plan.getLogicalPlan().getValue(LogicalPlan.GATEWAY_USE_SSL);
-    String gatewayUserName = plan.getLogicalPlan().getValue(LogicalPlan.GATEWAY_USER_NAME);
-    String gatewayPassword = plan.getLogicalPlan().getValue(LogicalPlan.GATEWAY_PASSWORD);
-    int timeout = plan.getLogicalPlan().getValue(LogicalPlan.PUBSUB_CONNECT_TIMEOUT_MILLIS);
-
-    if (gatewayAddress != null) {
-      try {
-        wsClient = new SharedPubSubWebSocketClient((gatewayUseSsl ? "wss://" : "ws://") + gatewayAddress + "/pubsub", timeout);
-        if (gatewayUserName != null && gatewayPassword != null) {
-          wsClient.setLoginUrl((gatewayUseSsl ? "https://" : "http://") + gatewayAddress + GATEWAY_LOGIN_URL_PATH);
-          wsClient.setUserName(gatewayUserName);
-          wsClient.setPassword(gatewayPassword);
-        }
+    try {
+      wsClient = new PubSubWSClientBuilder().setContext(plan.getLogicalPlan()).build();
+      if (wsClient != null) {
         wsClient.setup();
-      } catch (Exception ex) {
-        LOG.warn("Cannot establish websocket connection to {}", gatewayAddress, ex);
       }
+    } catch (Exception e) {
+      LOG.warn("Cannot establish websocket connection", e);
```

Comment:
Before there was information on connect URI, is it possible to keep this info available in the log message?

Reply:
I realized that but couldn't think of a cleaner way other than to maybe log the url within the builder at debug level, otherwise the manager needs to know how to get the address from the DAG. Also thought that if it is a url exception that would show up in the trace. What we will miss is the url during a setup exception. Would you be ok if I log the url in the builder in debug level.

Reply:
As the builder may return `null` and check for `null` is needed anyway, will it be better to catch all exceptions in the builder and log them there, so the builder can return `null`. In such case, only connect exception needs to be handled and you can get URI from wsClient.